### PR TITLE
chore: New api tests (FPLA-2910)

### PR DESCRIPTION
**Pull request by tomaszpowroznik**
https://tools.hmcts.net/jira/browse/FPLA-2910

### CODE REVIEW

File: service/src/functionalTest/java/uk/gov/hmcts/reform/fpl/StatelessCallbackApiTest.java [hunk 1]
```diff
@@ -1,58 +1,28 @@
...
 
-    @Autowired
-    private AuthenticationService authenticationService;
-
-    @Before
-    public void setUp() {
-        RestAssured.baseURI = testConfiguration.getFplUrl();
-        RestAssured.useRelaxedHTTPSValidation();
-    }
-
```
Comment (samuelzerozero) [inline service/src/functionalTest/java/uk/gov/hmcts/reform/fpl/StatelessCallbackApiTest.java:55]: as discussed, that can be flattened and inline in some stateless abstract class, with the only parameter the input/output to simplify (or something like that)

File: service/src/main/java/uk/gov/hmcts/reform/fnp/client/PaymentApi.java [hunk 2]
```diff
@@ -1,22 +1,33 @@
...
 public interface PaymentApi {
+
     @PostMapping("/credit-account-payments")
     CreditAccountPaymentRequest createCreditAccountPayment(
         @RequestHeader(AUTHORIZATION) String authorisation,
         @RequestHeader(SERVICE_AUTHORIZATION) String serviceAuthorization,
         @RequestBody CreditAccountPaymentRequest creditAccountPaymentRequest
     );
 
+    @GetMapping("/cases/{caseId}/payments")
```
Comment (samuelzerozero) [inline service/src/main/java/uk/gov/hmcts/reform/fnp/client/PaymentApi.java:26]: I know it's only for testing, at some point we need to look at the added contract tests we had for other stories. I'll create a ticket in the backlog

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/event/MessageJudgeEventData.java [hunk 3]
```diff
@@ -22,14 +23,15 @@
...
-        return new String[] {
+        return new String[]{
             "hasAdditionalApplications", "isMessageRegardingAdditionalApplications",
             "additionalApplicationsDynamicList", "relatedDocumentsLabel",
             "nextHearingLabel", "judicialMessageMetaData", "judicialMessageNote", "judicialMessageDynamicList",
             "messageJudgeOption", "judicialMessageReply", "hasJudicialMessages"
         };
     }
 
+    @JsonIgnore
```
Comment (samuelzerozero) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/model/event/MessageJudgeEventData.java:34]: was that a bug?

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseData.java [hunk 4]
```diff
@@ -702,6 +702,7 @@ public boolean hasFutureHearing(List<Element<HearingBooking>> hearingBookings) {
     }
 
     private final DocumentReference submittedForm;
+    private final DocumentReference draftApplicationDocument;
```
Comment (samuelzerozero) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseData.java:705]: interesting. was it never referred?

**Review by samuelzerozero — APPROVED** [review]

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/event/MessageJudgeEventData.java (hunk 3 above)
Comment (tomaszpowroznik) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/model/event/MessageJudgeEventData.java:34]: not a bug, but there is no such filed in ccd def, basically MessageJudgeEventData is never put into caseData. 
but in a test when empty CaseData is created and converted to CaseDetails then this filed appears, which is not intended

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseData.java (hunk 4 above)
Comment (tomaszpowroznik) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseData.java:705]: no, never, it is just put to caseDetails map, but never read back

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/event/MessageJudgeEventData.java (hunk 3 above)
Comment (samuelzerozero) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/model/event/MessageJudgeEventData.java:34]: ok good to know. a bit counterintuitive with @JsonUnwrapped, good to have @jsonIgnore

**Review by samuelzerozero — APPROVED** [review]

**Pull request merged**